tools/generate.py

```diff
@@ -601,6 +601,14 @@ def _gi_build_stub(
 
     # Functions
     for name in sorted(functions):
+        if hasattr(functions[name], "is_deprecated"):
+            deprecated = functions[name].is_deprecated()
```

Comment:
Looking at https://gitlab.gnome.org/GNOME/pygobject/-/blob/9b9f87d99f4f22681fbfaf22700e63c935c7bcd5/gi/pygi-cache.c#L721, you might be able to get more info using `.get_attribute("deprecated")`.

Reply:
I have added the check, unfortunately in the whole Gtk4 the attribute doesn't seems to be set

Reply:
I think that this information is only in the `gir` file and not in the `typelib` that pygobject uses. The only way to add this is to read the corresponding gir (for example `/usr/share/gir-1.0/Gtk-4.0.gir`) and extrapolate the `<doc-deprecated>` tag for the deprecated symbols

Reply:
Oh, right. The [commit](https://gitlab.gnome.org/GNOME/pygobject/-/commit/999679beaa9f5b36d9483abdbd30cd5e113b6bf6) that introduced the attribute code links to https://gitlab.gnome.org/GNOME/gobject-introspection/-/issues/80.

Reply:
I don't see any problem in reading another file during generation, they both should be installed simultaneously in the system anyway. I have pushed this change, please check if it is good enough for your use case

Reply:
Yeah, I think reading gir is fine. Especially when the stubs are already stored in the repo.